core/lib/zksync_core/src/l1_gas_price/gas_adjuster/mod.rs

```diff
@@ -63,23 +88,34 @@ impl GasAdjuster {
             .as_usize()
             .saturating_sub(1);
 
-        let last_processed_block = self.statistics.last_processed_block();
+        let last_processed_block = self.base_fee_statistics.last_processed_block();
 
         if current_block > last_processed_block {
-            // Report the current price to be gathered by the statistics module.
-            let history = self
-                .eth_client
-                .base_fee_history(
-                    current_block,
-                    current_block - last_processed_block,
-                    "gas_adjuster",
-                )
-                .await?;
+            let (base_fee_history, blob_base_fee_history) = Self::get_base_fees_history(
+                &self.eth_client,
+                (last_processed_block + 1)..=current_block,
+            )
+            .await?;
 
             METRICS
                 .current_base_fee_per_gas
-                .set(*history.last().unwrap());
-            self.statistics.add_samples(&history);
+                .set(*base_fee_history.last().unwrap());
+            self.base_fee_statistics.add_samples(&base_fee_history);
+
+            if let Some(current_blob_base_fee) = blob_base_fee_history.last() {
+                // Blob base fee overflows `u64` only in very extreme cases.
+                // It doesn't worth to observe exact value with metric because anyway values that can be used
+                // are capped by `self.config.max_blob_base_fee()` of `u64` type.
+                if current_blob_base_fee > &U256::from(u64::MAX) {
```

Comment:
why dont we use `bound_blob_base_fee`?

Reply:
Bounding also requires checking overflow. How I see it in my head: 
we will bound blob base fee much lower than `u64::MAX` and collect metric for all values that fit `u64::MAX`. So
- overflowing blob base fee bound -- extreme case, we use bounded for fee params and in eth sender and still report metric with actual base fee
- overflowing `u64` -- very extreme case, we still use bounded value, but don't report metric, also **important** that we do not panic